File: web/src/api/disk.ts
import request from '../utils/request';

export interface DiskStatus {
  total: number;
  used: number;
  free: number;
  used_pct: number;
  mysql_used: number;
  minio_used: number;
  backup_used: number;
  collected_at: string;
  interval: number; // seconds
}

export interface DiskTask {
  task_id: string;
  type: string;
  status: 'running' | 'success' | 'failed' | 'aborted';
  step: number;
  total: number;
  output: string;
  start_at: string;
}

export interface DirEntry {
  name: string;
  path: string;
  is_dir: boolean;
}

export const getDiskStatus = () =>
  request.get<{ code: number; data: DiskStatus }>('/disk/status');

export const setDiskInterval = (interval: number) =>
  request.put<{ code: number }>('/disk/interval', { interval });

export const browseFS = (path: string) =>
  request.get<{ code: number; data: DirEntry[] }>('/disk/fs', { params: { path } });

export const startMigrate = (target: 'mysql' | 'minio', newPath: string) =>
  request.post<{ code: number; data: DiskTask }>('/disk/migrate', { target, new_path: newPath });

export const getMigrateStatus = (taskId: string) =>
  request.get<{ code: number; data: DiskTask }>('/disk/migrate/status', { params: { task_id: taskId } });

export const changeBackupDir = (newPath: string) =>
  request.post<{ code: number; data: DiskTask }>('/disk/backup-dir', { new_path: newPath });

export const getBackupDirStatus = (taskId: string) =>
  request.get<{ code: number; data: DiskTask }>('/disk/backup-dir/status', { params: { task_id: taskId } });

File: server/model/disk.go
package model

import "time"

// DiskStatus 磁盘状态采集结果
type DiskStatus struct {
	Total       int64     `json:"total"`        // 字节
	Used        int64     `json:"used"`
	Free        int64     `json:"free"`
	UsedPct     float64   `json:"used_pct"`
	MySQLUsed   int64     `json:"mysql_used"`
	MinIOUsed   int64     `json:"minio_used"`
	BackupUsed  int64     `json:"backup_used"`
	CollectedAt time.Time `json:"collected_at"`
	Interval    int       `json:"interval"` // 秒
}

// DirEntry 目录浏览条目
type DirEntry struct {
	Name string `json:"name"`
	Path string `json:"path"` // 宿主机绝对路径
}

// MigrateRequest MySQL 或 MinIO 迁移请求
type MigrateRequest struct {
	Target  string `json:"target" binding:"required,oneof=mysql minio"` // "mysql" or "minio"
	NewPath string `json:"new_path" binding:"required"`
}

// BackupDirRequest 备份目录更换请求
type BackupDirRequest struct {
	NewPath string `json:"new_path" binding:"required"`
}

// DiskTask 磁盘操作任务状态
type DiskTask struct {
	TaskID  string `json:"task_id"`
	Type    string `json:"type"`     // migrate_mysql, migrate_minio, backup_dir
	Status  string `json:"status"`   // running, success, failed, aborted
	Step    int    `json:"step"`     // 当前步骤（1-based）
	Total   int    `json:"total"`    // 总步骤数
	Output  string `json:"output"`
	StartAt string `json:"start_at"`
}
